OneCommunityGlobal/HighestGoodNetworkApp
# yihan integrate team code into the Teams page

# Description
<img width="658" alt="截屏2023-09-06 09 22 32" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/81fd98d4-5a36-43ce-9394-cf826ca59dff">

This is the 3rd PR of "Create editable 4-letter team code option" task.

## Related PRS (if any):
This frontend PR is related to the [#529](https://github.com/OneCommunityGlobal/HGNRest/pull/529) backend PR.

## Main changes explained:
- Update file mockAdminState.js, mockStates.js, Team.test.js and allTeamsReducer.test.js for updating mock and unit test.
- Update file allTeamsAction.js and allTeamsReducer.js for adding teamcode parameter to updateTeam function.
- Update file TeamTable.jsx for integrating team code input box into table.
- Update file Reports.jsx for passing hasPermission and updateTeam function to TeamTable component.
- Update file DeleteTeamPopup.jsx, Team.jsx, TeamStatusPopup.jsx, Teams.jsx

## How to test:
1. check into current branch on both frontend and backend.
2. do `npm run start:local` to run this PR locally
3. log as Owner user or other uses who has Edit team 4-digit code permission.
4. go to Reports → Reports → Teams page
5. verify if you can edit team code of any team on that page. And the code you input should be auto-saved successfully.
